### ISSUE TITLE
Add more contrasts for training a multi-contrast model

### ISSUE THREAD

**Issue by naga-karthik** [ivadomed/model_sc_centerline#3]
Our experiment based on the preliminary hypothesis used on a single contrast (T2w) from the CanProCo dataset. It is important to gather the remaining contrasts for each subject and train a multi-channel/multi-contrast model (each additional contrast being added as an additional channel). This will be useful for robustness and also contrast-agnostic centerline detection. 

@valosekj Would you please be able to preprocess the remaining contrasts as we did for T2w?

**Comment by valosekj** [comment]
> @valosekj Would you please be able to preprocess the remaining contrasts as we did for T2w?

For remaining contrasts, we do not have ground truth spinal cord segmentation. Thus, we cannot easily obtain the centerline for these contrasts.

But I can co-register all contrasts to T2w. Then, we would be able to use the T2w centerline also for other contrasts.

**Comment by jcohenadad** [comment]
> For remaining contrasts, we do not have ground truth spinal cord segmentation. Thus, we cannot easily obtain the centerline for these contrasts.
> But I can co-register all contrasts to T2w. Then, we would be able to use the T2w centerline also for other contrasts.

It would be much easier to instead leverage our many other datasets on which we already have the spinal cord segmentations (spine-generic, etc.)

**Comment by naga-karthik** [comment]
> But I can co-register all contrasts to T2w. Then, we would be able to use the T2w centerline also for other contrasts.

Just for my understanding, this should also work right? I understand that it will take time but it should be good if we want to use all contrasts from CanProCo?

**Comment by jcohenadad** [comment]
> Just for my understanding, this should also work right? I understand that it will take time but it should be good if we want to use all contrasts from CanProCo?

yes, it should work, but registration needs to be QCed, which takes a bit of time

**Comment by valosekj** [comment]
> > Just for my understanding, this should also work right? I understand that it will take time but it should be good if we want to use all contrasts from CanProCo?
> 
> yes, it should work, but registration needs to be QCed, which takes a bit of time

I tried to co-register other contrasts to T2w based on nii headers using `-identity 1`:

https://github.com/ivadomed/model_sc_centerline/blob/cd6ffe1fba144bdb928b88dda68e48fac4ecc539/scripts/get_centerline_from_segmentation.sh#L150

To allow easy QC (we have >1000 coregistered images), I'm projecting the T2w centerline to the coregistered image using:

https://github.com/ivadomed/model_sc_centerline/blob/cd6ffe1fba144bdb928b88dda68e48fac4ecc539/scripts/get_centerline_from_segmentation.sh#L152-L155

Example (coregistered PSIR image and T2w_centerline):
![image](https://user-images.githubusercontent.com/39456460/216465636-5ad1306b-5508-4142-b6dd-b9fdfc011454.png)

Based on the initial quick QC assessment, it seems that the registration works well for most subjects, and also, it seems that this type of QC is reasonable!
